nautobot_golden_config/views.py

```diff
@@ -55,11 +56,25 @@ def extra_context(self):
 
     def alter_queryset(self, request):
         """Build actual runtime queryset as the build time queryset provides no information."""
-        qs = Device.objects.none()
-        for obj in models.GoldenConfigSetting.objects.all():
-            qs = qs | obj.get_queryset().distinct()
-
-        return self.queryset.filter(id__in=qs)
+        params = QueryDict()
+        params._mutable = True  # pylint: disable=protected-access
+        for obj in models.GoldenConfigSetting.objects.values_list("scope", flat=True).distinct():
+
+            params.update(QueryDict(urllib.parse.urlencode(obj, doseq=True)))
+        filtered_query = self.filterset(params, self.queryset)
+
+        return filtered_query.qs.annotate(
+            config_type=F("configcompliance__rule__config_type"),
+            backup_config=F("goldenconfig__backup_config"),
+            intended_config=F("goldenconfig__intended_config"),
+            compliance_config=F("goldenconfig__compliance_config"),
+            backup_last_success_date=F("goldenconfig__backup_last_success_date"),
+            intended_last_success_date=F("goldenconfig__intended_last_success_date"),
+            compliance_last_success_date=F("goldenconfig__compliance_last_success_date"),
+            backup_last_attempt_date=F("goldenconfig__backup_last_attempt_date"),
+            intended_last_attempt_date=F("goldenconfig__intended_last_attempt_date"),
+            compliance_last_attempt_date=F("goldenconfig__compliance_last_attempt_date"),
+        )
```

Comment:
Additionally can you please explain briefly why this `annotate` is required, again for posterity!